service/src/main/java/org/jboss/sbomer/processor/PncArtifactsToSbomPedigreeProcessor.java

```diff
@@ -15,50 +15,46 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.redhat.sbomer.processor;
+package org.jboss.sbomer.processor;
 
-import static org.redhat.sbomer.utils.Constants.SBOM_RED_HAT_BUILD_ID;
-import static org.redhat.sbomer.utils.Constants.DISTRIBUTION;
-import static org.redhat.sbomer.utils.Constants.SBOM_RED_HAT_ENVIRONMENT_IMAGE;
-import static org.redhat.sbomer.utils.SbomUtils.addExternalReference;
-import static org.redhat.sbomer.utils.SbomUtils.hasExternalReference;
-import static org.redhat.sbomer.utils.SbomUtils.setPublisher;
-import static org.redhat.sbomer.utils.SbomUtils.setSupplier;
-import static org.redhat.sbomer.utils.SbomUtils.addPedigreeCommit;
+import static org.jboss.sbomer.core.enums.ProcessorImplementation.PEDIGREE;
+import static org.jboss.sbomer.core.utils.Constants.DISTRIBUTION;
+import static org.jboss.sbomer.core.utils.Constants.SBOM_RED_HAT_BUILD_ID;
+import static org.jboss.sbomer.core.utils.Constants.SBOM_RED_HAT_ENVIRONMENT_IMAGE;
+import static org.jboss.sbomer.core.utils.SbomUtils.addExternalReference;
+import static org.jboss.sbomer.core.utils.SbomUtils.addPedigreeCommit;
+import static org.jboss.sbomer.core.utils.SbomUtils.hasExternalReference;
+import static org.jboss.sbomer.core.utils.SbomUtils.setPublisher;
+import static org.jboss.sbomer.core.utils.SbomUtils.setSupplier;
 
 import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
-import javax.validation.ValidationException;
 import javax.ws.rs.NotFoundException;
 
 import org.cyclonedx.model.Bom;
 import org.cyclonedx.model.Component;
 import org.cyclonedx.model.ExternalReference;
+import org.jboss.sbomer.core.utils.Constants;
+import org.jboss.sbomer.core.utils.RhVersionPattern;
+import org.jboss.sbomer.dto.ArtifactInfo;
+import org.jboss.sbomer.service.SBOMService;
 import org.jboss.util.Strings;
-import org.redhat.sbomer.utils.Constants;
-import org.redhat.sbomer.utils.RhVersionPattern;
-import org.redhat.sbomer.dto.ArtifactInfo;
-import org.redhat.sbomer.model.Sbom;
-import org.redhat.sbomer.service.SBOMService;
 
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
-@PncToSbomPedigree
+@Processor(PEDIGREE)
 @ApplicationScoped
 public class PncArtifactsToSbomPedigreeProcessor implements SbomProcessor {
 
     @Inject
     SBOMService sbomService;
 
     @Override
-    public Bom process(Sbom originalSbom) {
-        log.info("Applying SBOM_PEDIGREE processing to the SBOM: {}", originalSbom);
-
-        Bom originalBom = originalSbom.getCycloneDxBom();
-        if (originalBom == null) {
-            throw new ValidationException("Could not convert initial SBOM of build: " + originalSbom.getBuildId());
-        }
+    public Bom process(Bom originalBom) {
+        log.info(
+                "Applying SBOM_PEDIGREE processing to the SBOM: {}",
+                originalBom.getMetadata().getComponent().getPurl());
```

Comment:
My only concern here is that `metadata` is not a required field according to 1.4 spec: https://cyclonedx.org/docs/1.4/json/#metadata_component, so we are not sure this will always be populated (we haven't looked into other build type generators). So although when I changed the signature some days ago from Bom to Sbom I did not like it 100%, not sure if we should change it back _now_

Reply:
I think there was something as well behind the change -- inc case it will be moved to core/cli -- we would not depend at all on the entity.

Reply:
yes the entity is not used beside the Bom so this is the same reason why I did not like it 100%. ok so maybe we can verify if the metadata is not null and in case log other fields like `serialNumber` (but still not mandatory in the spec), or we could pick the first element in `components` (still not mandatory and not sure about the ordering priority).... but anyway this is a log and can be improved if needed